compiler/luci/pass/src/SubstitutePadV2ToPadPass.cpp

```diff
@@ -159,16 +259,39 @@ bool has_all_positive_values(luci::CircleConst *node)
   throw std::runtime_error("Not yet supported datatype");
 }
 
-// Check condition C3), C4) and C6)
-bool used_by_maxpool_only(luci::CirclePadV2 *)
+bool used_by_maxpool_only(luci::CircleNode *node, Paddings &paddings)
 {
-  //
-  // TODO Implement this
-  //
+  auto successor = get_only_succ(node);
+
+  // when successor is not only-succ
+  if (successor == nullptr)
+    return false;
+
+  if (auto maxpool = dynamic_cast<luci::CircleMaxPool2D *>(successor))
+  {
+    // Let's check condition C4)
+    return paddings.smaller_than(maxpool->filter()->h(), maxpool->filter()->w());
+  }
+
+  // Let's check condition C6)
+  if (auto transpose = dynamic_cast<luci::CircleTranspose *>(successor))
+  {
+    paddings.apply(transpose);
+    return used_by_maxpool_only(transpose, paddings);
+  }
+  // Support more 'reshaping op' later
 
   return false;
 }
 
+// Check condition C3), C4) and C6)
+bool used_by_maxpool_only(luci::CirclePadV2 *node)
```

Comment:
(optional) below function used `pad_v2` and `node` here. maybe go with one name?

Reply:
`node` was renamed to `pad_v2`. Thanks.